go-controller/pkg/util/ovs.go

```diff
@@ -766,3 +768,46 @@ func DetectSCTPSupport() (bool, error) {
 	}
 	return false, nil
 }
+
+// Finds any OVN Load Balancer based on external ID and value
+func FindOVNLoadBalancer(externalID, externalValue string) (string, string, error) {
+	out, stderr, err := RunOVNNbctl("--data=bare",
+		"--no-heading", "--columns=_uuid", "find", "load_balancer",
+		"external_ids:"+externalID+"="+externalValue)
+	if err != nil {
+		return "", stderr, err
+	}
+	return out, "", nil
+}
+
+// DetermineOVNTopoVersion determines what OVN Topology version is being used.
+func DetermineOVNTopoVersion() (int, error) {
+	ver := 0
+	stdout, stderr, err := RunOVNNbctl("--data=bare", "--no-headings", "--columns=name", "find", "logical_router",
+		fmt.Sprintf("name=%s", types.OVNClusterRouter))
+	if err != nil {
+		return ver, fmt.Errorf("failed in retrieving %s to determine the current version of OVN logical topology: "+
+			"stderr: %q, error: %v", types.OVNClusterRouter, stderr, err)
+	}
```

Comment:
for returning -1, I dont think it matters cause we are returning error. So the caller should always get an error and check against that. For 2nd point, yes we should use that instead of my randomly chosen big number :)